conftest.py

```diff
@@ -23,3 +30,82 @@ def pytest_addoption(parser):
 
 if sys.version_info < (3, 6):
     collect_ignore.append('pavement.py')
+
+
+@pytest.yield_fixture
+def global_tmpdir(request, tmpdir):
+    """
+    Return a temporary directory path object which is unique to each test
+    function invocation, created as a sub directory of the base temporary
+    directory. The returned object is a ``tests.lib.path.Path`` object.
+
+    This uses the built-in tmpdir fixture from pytest itself but modified
+    to return our typical path object instead of py.path.local as well as
+    deleting the temporary directories at the end of each test case.
+    """
+    assert tmpdir.isdir()
+    yield Path(str(tmpdir))
+    # Clear out the temporary directory after the test has finished using it.
+    # This should prevent us from needing a multiple gigabyte temporary
+    # directory while running the tests.
+    if not request.config.getoption("--keep-tmpdir"):
+        tmpdir.remove(ignore_errors=True)
+
+
+@pytest.fixture(autouse=True)
+def isolate(global_tmpdir):
+    """
+    Isolate our tests so that things like global configuration files and the
+    like do not affect our test results.
+
+    We use an autouse function scoped fixture because we want to ensure that
+    every test has it's own isolated home directory.
+    """
+
+    # TODO: Figure out how to isolate from *system* level configuration files
+    #       as well as user level configuration files.
+
+    # Create a directory to use as our home location.
+    home_dir = os.path.join(str(global_tmpdir), "home")
+    os.makedirs(home_dir)
+
+    # Create a directory to use as a fake root
+    fake_root = os.path.join(str(global_tmpdir), "fake-root")
+    os.makedirs(fake_root)
+
+    if sys.platform == 'win32':
```

Comment:
Hm... Do we need all this stuff with `USERPROFILE` and such?

Reply:
I'm not sure we do... Copy-pasting the whole function from pip's testsuite is a bit much, we certainly don't need those pip related changes to the environment (`PIP_DISABLE_PIP_VERSION_CHECK` and `PIP_REQ_TRACKER`).

Reply:
In fact, I think I can remove the `isolate` method entirely and just use `global_tmpdir`. What do you think?